xray/src/lib.rs

```diff
@@ -23,12 +23,27 @@ use cgmath::Point2;
 use fnv::FnvHashSet;
 use quadtree::{NodeId, Rect};
 use std::fs::File;
-use std::io::{Cursor, Read};
+use std::io::BufReader;
 use std::path::Path;
 
+pub use xray_proto_rust::proto;
+
 pub const CURRENT_VERSION: i32 = 2;
 
-#[derive(Debug)]
+pub fn read_meta_proto_from_file<P: AsRef<Path>>(filename: P) -> proto::Meta {
```

Comment:
nit: I prefer the newer way of writing this: `pub fn read_meta_proto_from_file(filename: impl AsRef<Path>) -> proto::Meta`. Both are equivalent in this case, but I find the newer to be easier to understand.